### PR TITLE
Tonyvanswet/pre commit

### PR DESCRIPTION
**Pull request by tonypls**
## Issue
Adds pre-commit prettifying and tests.

## Description
We had shifted to running pre push but this is too late if prettier or eslint needs to add and changes. Also now run vitest on any changed files.

### CODE REVIEW

File: web/package.json
Comment (VIKTORVAV99) [inline web/package.json:173]: Why do we need this? Can't we run prettier directly?
I'm pretty sure it only format changed files anyway.

(We did run prettier as a pre commit before I think?)

Comment (tonypls) [reply web/package.json:173]: I think pre rewrite we did but it hasn't worked in a while.

Prettier suggested using this [here](https://prettier.io/docs/en/precommit.html) and it's just a dev dep

Comment (VIKTORVAV99) [reply web/package.json:173]: Don't we already do option 1 described there?

Comment (tonypls) [reply web/package.json:173]: I assumed a library would have some optimisations, I timed the execution twice here are the results:
pretty-quick --staged : 
0.82s user 0.33s system 103% cpu 1.111 total
0.80s user 0.30s system 99% cpu 1.108 total

prettier -w . :
6.20s user 0.60s system 159% cpu 4.263 total
6.21s user 0.58s system 159% cpu 4.247 total

Not sure about the backwards math of the bash time function 🤔  but pretty-quick was pretty quick 🤣

Comment (VIKTORVAV99) [reply web/package.json:173]: did you try prettier with the `--cache` option as well?

Comment (tonypls) [reply web/package.json:173]: 1.29s user 0.35s system 115% cpu 1.423 total
1.30s user 0.33s system 116% cpu 1.394 total

:D pretty good

Comment (VIKTORVAV99) [reply web/package.json:173]: Thanks for testing it!
It does looks like pretty-quick still is faster so I'll leave it up to you but the less dependencies the better in my opinion. 😅

**Review by madsnedergaard — APPROVED** [review]
LGTM once merge conflict on version number is solved :)

**Pull request merged**
